### PR TITLE
Add upload_csv route

### LINKED ISSUE #67: Add upload_csv
Based on the documentation here:

https://dune.com/docs/api/api-reference/upload-data/?h=data+upload#endpoint

### PR DESCRIPTION
**Pull request by bh2smith**
This PR adds CSV upload functionality to the API client.

Unfortunately, it appears, I have the ability to completely overwrite other users tables... Not sure if this was intended. cc @msf 

Tests are failing, because I assumed I would get a False success response when trying to overwrite someone else's table.

Closes #67

Remaining Question: Do we also was to update the async client? If so, it would probably need more work since it has not been kept updated like the other.

### CODE REVIEW

File: dune_client/base_client.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
 from typing import Dict
 
 
-# pylint: disable=too-few-public-methods
```
Comment (bh2smith) [inline dune_client/base_client.py:12]: This is no longer needed (thought I would sneak it in here).

**Comment by dsalv** [comment]
> This PR adds CSV upload functionality to the API client.
> 
> Unfortunately, it appears, I have the ability to completely overwrite other users tables... Not sure if this was intended. cc @msf
> 
> Tests are failing, because I assumed I would get a False success response when trying to overwrite someone else's table.
> 
> Closes #67
> 
> Remaining Question: Do we also was to update the async client? If so, it would probably need more work since it has not been kept updated like the other.

The ability to overwrite other people's tables under `dune_upload` is the intended behavior. Tables cannot be overwritten if they are under users' personal namespaces (e.g. dune.bh2smith.dataset_{table_name}). We introduced personal namespaces last week. I am not sure that would be helpful in your tests

File: dune_client/client.py [hunk 2]
```diff
@@ -374,3 +374,26 @@ def make_public(self, query_id: int) -> None:
...
+        The write API allows you to upload any .csv file into Dune. The only limitations are:
+
+        - File has to be < 200 MB
+        - Column names in the table can't start with a special character or digits.
+
+        Below are the specifics of how to work with the API.
+        """
+        response_json = self._post(
+            route="/table/upload/csv",
+            params={
```
Comment (msf) [inline dune_client/client.py:390]: @dsalv how do we specify the namespace the table should be in?

Comment (dsalv) [reply dune_client/client.py:390]: right now it is not possible to customize the namespace. All publicly visible tables go as `dune_upload.{table_name}`. All tables are also duplicated as `dune.{username}.dataset_{table_name}`.

The current plan is to keep the personal namespaces and deprecate `dune_upload` soon. There is no plan to support customizable namespaces at the moment. This needs to be escalated if we think customizable namespaces are critical

**Comment by bh2smith** [comment]
So far, it looks like just commentary on intended functionality. At the moment the implementation and tests are aligned with the intended functionality. If its not blocking, I'd think we can merge as is and adjust later as necessary. 

Please let me know if there is anything that should be changed here and feel free to approve if all is good.

**Review by msf — APPROVED** [review]
lgtm

**Review by dsalv — APPROVED** [review]

**Pull request merged**